app/index.js

```diff
@@ -1,832 +1,260 @@
-        sendDataToCompanion(_feedbackData);
-    }
+function showFace(flowback = false, isFirst = false) {
+    let skipQuestion = false;
 
-    function locationError(error) {
-        console.log("GPS location fail");
-        console.log(error);
-        _feedbackData.lat = null;
-        _feedbackData.lon = null;
-        sendDataToCompanion(_feedbackData);
-    }
-}
-
-function sendDataToCompanion(data) {
-    console.log("Sending feedback data ... ");
-
-    if (JSON.stringify(data).length > (messaging.peerSocket.MAX_MESSAGE_SIZE - 200)) {
-        console.log('The message you are sending has a length of : ' + JSON.stringify(data).length + "but you can only send messages up to this size" + messaging.peerSocket.MAX_MESSAGE_SIZE);
-    }
+    if(isFirst) {
+        currentView=0;
+    }    
 
-    if (messaging.peerSocket.readyState === messaging.peerSocket.OPEN) {
+    // go through all views and set to none
+    allViews.map((v) => {
+        v.style.display = "none";
+    });
 
-        console.log("data sent via Peer Socket");
-        messaging.peerSocket.send(data);
+    jsonFlow.style.display = "inline";
 
-        if (!production) {
-            voteLogPeerTransfer++;
-            voteLogPeerTransferLabel.text = voteLogPeerTransfer + "pt;";
-        }
-    } else {
-        console.log("No peerSocket connection. Attempting to send via file transfer");
-
-        let local_file;
-
-        // try to read file with local data
-        try {
-            console.log("checking if local file exists");
-            local_file = fs.readFileSync("local.txt", "json");
-        } catch (e) {
-            // if can't read set local file to empty
-            console.log("creating empty local.txt file");
-            local_file = [];
-        }
-
-        // push new reponce and save
-        local_file.push(data);
-
-        fs.writeFileSync("local.txt", local_file, "json");
-
-        // Prepare outbox for file transfer
-        outbox
-            .enqueueFile("local.txt")
-            .then((ft) => {
-                console.log(`Transfer of ${ft.name} successfully queued.`);
-
-                // Let user know that data is in queue
-                storageLabel.text = `q ${local_file.length}`;
-                // On change of ft, launch file transfer event
-                ft.onchange = onFileTransferEvent;
-            })
-            .catch((error) => {
-                console.log(`Failed to schedule transfer: ${error}`);
-                storageLabel.text = `e ${local_file.length}`;
-            })
+    //Does current flow have any requirements?
+    if (questionsFlow[currentView].requiresAnswer.length !== 0) {
+        //if so, see if the current feedback meets those requirements
+        questionsFlow[currentView].requiresAnswer.map((req) => {
+            if (feedbackData[req.question] !== req.value) {
+                //requirements not met, skipping question
+                skipQuestion = true;
+            }
```

Comment:
This is related to the code that Lionel wrote to fix the issue with the buzz function. The downside of moving the code related to the buzz was that we did not have access to the variable "current view". Consequently when was time to buzz to remind the user to complete the survey we did not know whether the user was or was not completing a survey. He, therefore, exported the functions getView and showFace so they can be accessed by "buzz.js" but this is causing this error. I will look into it but since the error message does not really tell us what is going on, it will be hard to find the root cause of the problem. Maybe I will have to revert his changes.

Reply:
@pjayathissa I have spent a couple of hours on this but I was not able to solve the issue. I am sure it is something super simple since basically we just need to share a variable and a function between two files, but unfortunately I cannot solve the issue. @LionelKarlen kindly let us know if you are able to fix the error.

Reply:
Yeah, that's the error I kept getting. It seems fitbit uses some weird form of es6 where neither commonjs imports nor es6 imports work correctly and I have no idea why it's happening, sorry. Maybe to help debug: I was able to use default es6 exports, but it's not designed for multiple objects. I bypassed this by exporting an object that contains showFace() and getCurrentView(), I believe the error has something to do with this bypass. Normally, in es6, you can just export any object and then import it with `import { module } from 'path'` but whenever I tried that it didn't even run the flow code anymore, hence bypassing default exports.

Reply:
Also, turns out you cannot just import variables because it violates the scope, afaik. Hence getView() instead of just the variable currentView. This is also the reason I added an optional parameter to showFace(), so that I can change the value of currentView in the same scope it was initialised in.

Reply:
@pjayathissa what shall we do? I am happy to keep Lionel changes even if there is that small error since the clock face is working. However, if you are concerned about that error, we will have to either revert the latest changes, or bring the buzz function back into index.js

Reply:
I'm ok to have the small error in `coziedev`, but it would need to be solved before we bring it to `master`

Reply:
To be honest I do not know how to fix this issue and I do not have much time to work on it at the moment, consequently I would not push it to `coziedev` if we do not want to bring it into `master`.